Plugin update manager: empty state when no eligible plugins are installed

TimBroddin opened the pull request:
Closes https://github.com/Automattic/wp-calypso/issues/88404

## Proposed Changes

- Checks whether there are eligble plugins installed inside `<PluginsUpdateManager /> ` and sets this on the context
- Shows a message + CTA on the list screen informing the user that they have no eligble plugins
- Minor refactorings

![CleanShot 2024-03-18 at 13 41 38@2x](https://github.com/Automattic/wp-calypso/assets/528287/85819737-2288-4be6-90d2-f7a50a62c726)


## Testing Instructions

1. Apply this PR
2. Check whether you see the message when no eligble plugins are installed (you can mock this by returning `[]` inside the select function of `client/data/plugins/use-core-plugins-query.ts`.
3. Tets for regressions


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

TimBroddin committed:
Plugin update manager: empty state when no eligible plugins are installed

JanaMW27 commented:
Nice one!

A couple of thoughts:
- Let's remove the + from the CTA
- In the sentence I would replace "manage" with "update" since this section is all about update schedules rather than managing the plugin per se.

TimBroddin committed:
Remove +

bogiii commented on client/blocks/plugins-update-manager/index.tsx:
```diff
@@ -40,6 +41,14 @@ export const PluginsUpdateManager = ( props: Props ) => {
+		true
+	);
+	const siteHasEligiblePlugins =
+		isPluginsFetched && plugins.filter( ( plugin ) => ! plugin.is_managed ).length > 0;
```
This filtering has already been done using `useCorePluginsQuery`; see the second `true` parameter. It is for filtering managed plugins.

TimBroddin committed:
Change copy

TimBroddin commented:
> Nice one!
> 
> A couple of thoughts:
> 
> * Let's remove the + from the CTA
> * In the sentence I would replace "manage" with "update" since this section is all about update schedules rather than managing the plugin per see.

I removed the + and changed the copy. Thanks for the review @JanaMW27 

![CleanShot 2024-03-18 at 14 41 43@2x](https://github.com/Automattic/wp-calypso/assets/528287/4dcb316d-2365-4f10-9dc3-81ef684cba60)

bogiii commented on client/blocks/plugins-update-manager/hooks/use-site-has-eligible-plugins.ts:
```diff
@@ -1,18 +1,9 @@
-	return siteHasEligiblePlugins;
+	const { siteHasEligiblePlugins, siteHasEligiblePluginsLoading } = useContext(
+		PluginUpdateManagerContext
+	);
```
From my point of view, it looks like an overhead setting and getting this data from the context.

We can use it directly from the `useCorePluginsQuery`; it persists and shares the info between components, so we can have the right data if it's loading and the plugin data.

What do you think?

TimBroddin replied:
> From my point of view, it looks like an overhead setting and getting this data from the context.
> 
> We can use it directly from the `useCorePluginsQuery`; it persists and shares the info between components, so we can have the right data if it's loading and the plugin data.
> 
> What do you think?

I refactored the logic:
- Moved all the logic to `useSiteHasEligiblePlugins` and removed setting it on the context
- Allow passing the siteSlug to `useSiteHasEligiblePlugins` for usage inside the root `<PluginsUpdateManager />` component

bogiii replied:
Thanks for handling it. Nice work. 👍

TimBroddin committed:
Refactor

TimBroddin committed:
Case where there are schedules added, but there are no eligble plugins at the current moment

bogiii commented on client/blocks/plugins-update-manager/schedule-list.tsx:
```diff
@@ -52,8 +55,15 @@ export const ScheduleList = ( props: Props ) => {
+		isLoadingSchedules ||
+		isLoadingCanCreateSchedules ||
+		isEligibleForFeatureLoading ||
+		siteHasEligiblePluginsLoading;
```
I like it; it's more readable. 👍

bogiii reviewed (approved):
Nice work! No regression were found. 🚢
